lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -65,3 +69,94 @@ function dosomething_kudos_get_kudos_query($params = [], $count = NULL, $start =
   $result = $query->execute();
   return array_keys($result['kudos']);
 }
+
+
+/**
+ * Sort kudos collection based on taxonomy term and then group them by term.
+ *
+ * @param array $data Collection of individual kudos items.
+ *
+ * @return array
+ */
+function dosomething_kudos_sort($data) {
+  // The array containing collection of kudos items passed to this function
+  // consists of individual kudos items. However, within this collection
+  // there could be multiple kudos items that all fall within the same
+  // taxonomy term. So the following series of functions sort and then
+  // group the kudos based on taxonomy term, along with providing a total
+  // count of kudos for each term.
+
+  usort($data, 'dosomething_kudos_sort_by_taxonomy_term');
+
+  $data = dosomething_kudos_group_by_taxonomy_term($data);
+  $data = dosomething_kudos_get_totals_by_taxonomy_term($data);
+  return $data;
+}
+
+
+/**
+ * Sort kudos collection by taxonomy term id using a value comparison function.
+ *
+ * @param array $a Single kudos object.
```

Comment:
Just clarifying: Kudos object == an instance of the `Kudos` class?  (One could also read this as a result row in `dosomething_kudos`... a Kudos taxonomy term)